subsys/mpsl/cx/nrf700x/mpsl_cx_nrf700x.c

```diff
@@ -219,9 +219,19 @@ static uint32_t req_grant_delay_get(void)
 
 static int32_t register_callback(mpsl_cx_cb_t cb)
 {
+	int32_t ret = 0;
 	callback = cb;
 
-	return 0;
+	if (cb != NULL) {
+		ret = gpio_pin_interrupt_configure_dt(&grant_spec,
+			GPIO_INT_ENABLE | GPIO_INT_EDGE | GPIO_INT_EDGE_BOTH);
+	}
+	else
+	{
```

Comment:
compliance check fail, please try:
```suggestion
	} else {
```
and in other places as well